wdl-lint/src/v1/preamble_whitespace.rs

```diff
@@ -0,0 +1,192 @@
+
+    fn version_statement(
+        &mut self,
+        state: &mut Self::State,
+        reason: VisitReason,
+        stmt: &VersionStatement,
+    ) {
+        if reason == VisitReason::Exit {
+            return;
+        }
+
+        // We're finished after the version statement
+        self.finished = true;
+
+        // If the previous token is whitespace and its previous token is a comment,
+        // then the whitespace should just be two lines
+        match stmt
+            .syntax()
+            .prev_sibling_or_token()
+            .and_then(SyntaxElement::into_token)
+        {
+            Some(whitespace) if whitespace.kind() == SyntaxKind::Whitespace => {
+                let range = whitespace.text_range();
+                if whitespace
+                    .prev_sibling_or_token()
+                    .map(|s| s.kind() == SyntaxKind::Comment)
+                    .unwrap_or(false)
+                {
+                    // There's a preceding comment, so it is expected that the whitespace is only
+                    // two newlines in a row
+                    let text = whitespace.text();
+                    let mut count = 0;
+                    for (line, start, next_start) in lines_with_offset(text) {
+                        // If this is not a blank line, it's unnecessary whitespace
+                        if !line.is_empty() {
+                            state.add(unnecessary_whitespace(Span::new(
+                                usize::from(range.start()) + start,
+                                line.len(),
+                            )));
+                        }
+
+                        count += 1;
+                        if count == 2 {
+                            // If the previous byte isn't a newline, then we are missing a blank
+                            // line
+                            if text.as_bytes()[next_start - 1] != b'\n' {
+                                state.add(expected_blank_line(Span::new(
+                                    usize::from(range.start()) + start,
+                                    1,
+                                )));
+                            }
+
+                            // We're at the second line, the remainder is unnecessary whitespace
+                            if next_start < text.len() {
+                                state.add(unnecessary_whitespace(Span::new(
+                                    usize::from(range.start()) + next_start,
+                                    text.len() - next_start,
+                                )));
+                            }
+
+                            break;
+                        }
+                    }
+
+                    // We expected two lines
+                    if count < 2 {
+                        state.add(expected_blank_line(range.to_span()));
+                    }
+                } else {
+                    // Whitespace without a comment before it
```

Comment:
This is another case that isn't clear to me.

Reply:
This case can be hit in two ways.

The first is that there's a single whitespace token before the version statement:

```wdl
  
  
version 1.1
```

In the above case, the entire whitespace token is unnecessary.

The other is a parse error as only a comment token would normally precede a whitespace token in the preamble:

```wdl
task oops {
}

version 1.1
```

Here we actually end up emitting an error that a version statement must come before the task and the remainder of the file, including the version statement, is treated as "unparsed"; it would subsequently not be linted.